### PR TITLE
fix: add predict.py & analyze.py

### PR DESCRIPTION
**Pull request by yan91083**

### CODE REVIEW

File: experimental/eval/predict.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+            "download",
+            "--model",
+            MODEL_ID,
+        ]
+    )
+
+
+image = (
+    Image.from_registry(
+        "tabbyml/tabby:0.5.4",
```
Comment (wsxiaoys) [inline experimental/eval/predict.py:32]:
```suggestion
        "tabbyml/tabby:0.5.5",
```

File: experimental/eval/predict.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    .run_function(download_model)
+)
+
+stub = Stub("tabby-" + MODEL_ID.split("/")[-1], image=image)
+
+
+@stub.cls(
+    gpu=GPU_CONFIG,
+    concurrency_limit=10,
+    allow_concurrent_inputs=1,
```
Comment (wsxiaoys) [inline experimental/eval/predict.py:49]:
```suggestion
    allow_concurrent_inputs=4,
```

**Review by wsxiaoys — CHANGES_REQUESTED** [review]

File: experimental/eval/predict.py [hunk 3]
```diff
@@ -8,15 +8,17 @@
 import asyncio
 
 GPU_CONFIG = gpu.A10G()
-#MODEL_ID =  os.environ.get("MODEL_ID", "abc")
-MODEL_ID = "TabbyML/StarCoder-7B"
+#MODEL_ID =  os.environ.get("MODEL_ID", "")
```
Comment (koalaxu) [inline experimental/eval/predict.py:11]: Remove commented out code?

File: experimental/eval/predict.py [hunk 4]
```diff
@@ -8,15 +8,17 @@
...
 LAUNCH_FLAGS = ["serve", "--model", MODEL_ID, "--port", "8000", "--device", "cuda"]
 #print(f'MODEL_ID = `{MODEL_ID}`')
 
-
 def download_model():
     import subprocess
-    print(f'Loading model `{MODEL_ID}`')
+    import os
+    MODEL_ID = os.popen("cat /tmp/tabby_model_id").read().strip()
+    print(f'MODEL_ID={MODEL_ID}')
```
Comment (koalaxu) [inline experimental/eval/predict.py:21]: Remote model ID

File: experimental/eval/predict.py [hunk 5]
```diff
@@ -60,6 +63,9 @@ def __enter__(self):
 
         my_env = os.environ.copy()
         my_env["TABBY_DISABLE_USAGE_COLLECTION"] = "1"
+        MODEL_ID = os.popen("cat /tmp/tabby_model_id").read().strip()
+        print(f'MODEL_ID={MODEL_ID}')
```
Comment (koalaxu) [inline experimental/eval/predict.py:67]: Local model ID

**Pull request closed without merging**